File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __traverse(self, node: Optional[TreeNode], solution: List[TreeNode]) -> None:
        if node is None: 
            return
        self.__traverse(node.left, solution)
        solution.append(node.val)
        self.__traverse(node.right, solution)


    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        solution = []
        self.__traverse(root, solution)
        return solution

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    void traverse(TreeNode* node, vector<int>& solution) {
        if (!node) return;
        traverse(node->left, solution);
        solution.push_back(node->val);
        traverse(node->right, solution);
    }

public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> solution;
        traverse(root, solution);
        return solution;
    }
};